tests/utils.py

```diff
@@ -32,7 +38,40 @@
 
 
 try:
-    from ._openmp_test_helper import check_openmp_num_threads  # noqa: F401
+    import tests._openmp_test_helper.openmp_helpers_inner  # noqa: F401
     cython_extensions_compiled = True
 except ImportError:
     cython_extensions_compiled = False
+
+
+def threadpool_info_from_subprocess(code):
+    """Utility to call threadpool_info in a subprocess
+
+    `code` is exectuted before calling threadpool_info
+    """
+    handle, filename = tempfile.mkstemp(suffix='.py')
+    os.close(handle)
+
+    src = code + textwrap.dedent("""
+    from threadpoolctl import threadpool_info
+    print(threadpool_info())
+    """)
+
+    path1 = normpath(dirname(threadpoolctl.__file__))
+    path2 = os.path.join(path1, "tests", "_openmp_test_helper")
+    pythonpath = os.pathsep.join([path1, path2])
+    env = os.environ.copy()
+    try:
+        env["PYTHONPATH"] = os.pathsep.join([pythonpath, env["PYTHONPATH"]])
+    except KeyError:
+        env["PYTHONPATH"] = pythonpath
+
+    try:
+        with open(filename, "wb") as f:
+            f.write(src.encode("utf-8"))
+        cmd = [sys.executable, filename]
```

Comment:
I think you could have used `cmd = [sys.executable, "-m", "threadpoolctl" + "-i" + module_to_import]`
and then read the result with `json.loads`.

Reply:
much simpler indeed :)